### DISCUSSION TITLE
Unable to code the ESP01

### DISCUSSION THREAD

**Discussion by sQur1** [micropython/micropython#10747] [ESP8266]
I have no problem uploading the code using these connections ([image](https://www.diyhobi.com/wp-content/uploads/2017/07/TTL-CH340G-Converter-Module-Adapter-Esp-01-wirings-1024x597.jpg) but when I try to code the board, I remove the GPIO0 to GND cable from my ESP01 which is what I assume is the correct way to change the board's mode from Bootloader to normal mode but it still doesn't seem to work. I've tried with the 1.19.1 and 1.18 (tried the 1MB and 512K versions through 115200 baudrate) firmware, but the REPL in Thonny always shows the board as busy. I tried unplugging and replugging the device, CTRL+C doesn't do anything... Help please

EDIT: I tried the RST by touching it to GND. Im able to see that it's detecting a reset in the REPL of Thonny, but I'm still unable to get through pass the "Device is busy" message.

**Comment by robert-hh** [comment]
Does the board not work at all or does is reboot over and over. Using differenf flash modes like DOUT may help as well.
Try to use another connection tool than Thonny. Thonny hides to much of what is going on. On Windows, try e.g. Putty. With OS X or Linux try screen or picocom.

**Reply by sQur1** [reply]
The board works because I'm able to flash/erase firmware without issues. I've tried with VSCode and it still says that the device is busy.

**Reply by robert-hh** [reply]
Even if the flash mode is wrong, you can flash & erase it. Try something Basic for communication instead of Thonny. Which COM devices are listed in the device manager, and which one do you use? Which PC operating system?

**Reply by cve2022** [reply]
> On Windows, try e.g. Putty. With OS X or Linux try screen or picocom.

For the Putty addicted 😉, it is available for Linux as well. Apparently, for Mac OS too, according to a quick duckduckgo search.

**Reply by sQur1** [reply]
> Even if the flash mode is wrong, you can flash & erase it. Try something Basic for communication instead of Thonny. Which COM devices are listed in the device manager, and which one do you use? Which PC operating system?

I'm using MacOS so I don't have Device manager. Is there a Mac alternative? I'm very new to electronics so I don't know how to operate a more basic communication software for COM. I know only Thonny. What would be the point of using a more basic one? Would it give away the reason why it isn't able to reach the board?

**Reply by robert-hh** [reply]
The devices can be seen in the directory /dev. Try `ls /dev/tty*` from the command line with and without the board connected. The use `screem <device_path>` to conect, or use `mpremote <device_path>` to connect. Both use 115200 baud as default speed.

Edit: Using my historic Mac I cannot connect a ESP board with a CH340 chip. The Mac restarts immediately. Something wrong with the driver. I can however connect e.g. a RP2 pico board or boards with a CP2102 chip.
